spec/features/applications/new_spec.rb

```diff
@@ -45,11 +45,11 @@
       click_link("Apply for Pets")
 
       expect(current_path).to eq("/applications/new")
-      expect(page).to have_css("#pet_id-#{@pet_1.id}")
-      expect(page).to have_css("#pet_id-#{@pet_2.id}")
-      expect(page).to_not have_css("#pet_id-#{@pet_3.id}")
+      expect(page).to have_css("#checkbox-#{@pet_1.id}")
+      expect(page).to have_css("#checkbox-#{@pet_2.id}")
+      expect(page).to_not have_css("#checkbox-#{@pet_3.id}")
 
-      check "pet_id-#{@pet_1.id}"
+      within("#checkbox-#{@pet_1.id}") { check "applied_for_" }
```

Comment:
instead of trying to target unique checkbox fields, those unique paragraphs are targeted instead